src/datepicker/DateRangePickerState.ts

```diff
@@ -0,0 +1,173 @@
+/**
+ * All credit goes to [React Spectrum](https://github.com/adobe/react-spectrum)
+ * We improved the Calendar from Stately [useDatePickerState](https://github.com/adobe/react-spectrum/blob/main/packages/%40react-stately/datepicker/src/useDatePickerState.ts)
+ * to work with Reakit System
+ */
+
+import {
+  Validation,
+  FocusableProps,
+  ValueBase,
+  RangeValue,
+  ValidationState,
+} from "@react-types/shared";
+import * as React from "react";
+import { useCompositeState } from "reakit";
+import { useControllableState } from "@chakra-ui/hooks";
+
+import {
+  parseDate,
+  stringifyDate,
+  parseRangeDate,
+  isInvalidDateRange,
+} from "../utils";
+import { useSegmentState } from "../segment";
+import { makeRange } from "../calendar/__utils";
+import { useRangeCalendarState } from "../calendar";
+import { DateTimeFormatOpts, RangeValueBase } from "../utils/types";
+import { PickerBaseInitialState, usePickerBaseState } from "../picker-base";
+
+export interface DateRangePickerInitialState
+  extends PickerBaseInitialState,
+    Validation,
+    FocusableProps,
+    ValueBase<RangeValue<string>>,
+    RangeValueBase<string> {
+  placeholderDate?: string;
```

Comment:
So we can get the placeholderDate as well in string.

If yes you can change this in DatePickerState too.